demos/http/http_demo_plaintext/http_demo_plaintext.c

```diff
@@ -0,0 +1,552 @@
+                           &( ( struct sockaddr_in * ) pServerInfo )->sin_addr,
+                           resolvedIpAddr,
+                           sizeof( resolvedIpAddr ) );
+            }
+
+            LogInfo( ( "Attempting to connect to resolved IP address: %s.",
+                       resolvedIpAddr ) );
+
+            returnStatus = connect( *pTcpSocket, pServerInfo, serverInfoLength );
+
+            if( returnStatus == -1 )
+            {
+                close( *pTcpSocket );
+            }
+            else
+            {
+                LogInfo( ( "Connected to IP address: %s.",
+                           resolvedIpAddr ) );
+                break;
+            }
+        }
+
+        if( pIndex == NULL )
+        {
+            /* Fail if no connection could be established. */
+            returnStatus = EXIT_FAILURE;
+            LogError( ( "Could not connect to any resolved IP address from %.*s.\n",
+                        ( int ) strlen( pServer ),
+                        pServer ) );
+        }
+        else
+        {
+            returnStatus = EXIT_SUCCESS;
+            LogInfo( ( "TCP connection established with %.*s.\n",
+                       ( int ) strlen( pServer ),
+                       pServer ) );
+        }
+    }
+    else
+    {
+        LogError( ( "Could not resolve host %.*s.\n",
+                    ( int ) strlen( pServer ),
+                    pServer ) );
+        returnStatus = EXIT_FAILURE;
+    }
+
+    /* Set the socket option for send and receive timeouts. */
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        transportTimeout.tv_sec = 0;
+        transportTimeout.tv_usec = ( TRANSPORT_SEND_RECV_TIMEOUT_MS * 1000 );
+
+        /* Set the receive timeout. */
+        if( setsockopt( *pTcpSocket,
+                        SOL_SOCKET,
+                        SO_RCVTIMEO,
+                        ( char * ) &transportTimeout,
+                        sizeof( transportTimeout ) ) < 0 )
+        {
+            LogError( ( "Setting socket receive timeout failed." ) );
+            returnStatus = EXIT_FAILURE;
+        }
+
+        /* Set the send timeout. */
+        if( setsockopt( *pTcpSocket,
+                        SOL_SOCKET,
+                        SO_SNDTIMEO,
+                        ( char * ) &transportTimeout,
+                        sizeof( transportTimeout ) ) < 0 )
+        {
+            LogError( ( "Setting socket send timeout failed." ) );
+            returnStatus = EXIT_FAILURE;
+        }
+    }
+
+    if( pListHead != NULL )
+    {
+        freeaddrinfo( pListHead );
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static int32_t transportSend( HTTPNetworkContext_t * pContext,
+                              const void * pBuffer,
+                              size_t bytesToSend )
+{
+    int32_t bytesSend = 0;
+
+    bytesSend = send( ( int ) pContext->tcpSocket, pBuffer, bytesToSend, 0 );
```

Comment:
Is the `int` cast required? Isn't the `tcpSocket` already of `int` type?

Reply:
hmm not sure why it's casted either